### PR TITLE
Feature/map

### PR DESCRIPTION
**Pull request by egrace479**
`components/prototype-map.py` generates interactive map of the specimen locations colored by subspecies, with size of bubbles proportional to the number of specimens of that subspecies at that locality.

Renames `test_components` folder as `components` with addition of new `graphs.py`.

Step one in larger integration of mapping feature and code redundancy reduction.

### CODE REVIEW

File: components/graphs.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    '''
+    Generates interactive graph of Species and Subspecies by location.
+    
+    Parameters:
+    -----------
+    df - dataframe of specimens
+
+    Returns: 
+    --------
+    fig - Map of their locations
```
Comment (hlapp) [inline components/graphs.py:15]: Is it clear what type the returned "Map" object is going to be, or would it be worth saying?

**Review by hlapp — APPROVED** [review]
See inline comment for question / suggestion.

File: components/graphs.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    '''
+    fig = px.scatter_geo(df,
+                        lat = df.lat,
+                        lon = df.lon,
+                        projection = "natural earth",
+                        hover_data = ["Species", "Subspecies"],
+                        size = df.samples_at_locality,
+                        color = "Subspecies",
+                        color_discrete_sequence = px.colors.qualitative.Bold)
+    fig.update_geos(fitbounds = "locations",
```
Comment (thompsonmj) [inline components/graphs.py:25]: I would vote to add `showcountries=True, countrycolor="Grey",` here.

**Review by thompsonmj — APPROVED** [review]
I think country boarders would be useful on the map!

File: components/graphs.py (hunk 1 above)
Comment (egrace479) [reply components/graphs.py:15]: It's just a Plotly figure object, so I think the variable name `fig` is sufficient (and reasonably standard).

**Review by johnbradley — APPROVED** [review]
LGTM

**Pull request merged**